### ISSUE TITLE
'perldoc -oman' is incorrectly rendered under Cygwin.

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#7542]
Migrated from [rt.perl.org#31997](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=31997) (status was 'resolved')

Searchable as RT31997$

**Comment by p5pRT** [comment]
### From @dolmen

This is a bug report for perl from dolmen@&#8203;users\.sourceforge\.net\,
generated with the help of perlbug 1\.35 running under perl v5\.8\.5\.

\(but manually sent by email\)



---
'perldoc \-oman' is incorrectly rendered under Cygwin\.
I tried with the following terminals&#8203;: cygwin\, xterm\, rxvt\.

The problem is that the nroff output is piped to the default
pager which is not defined for cygwin\.
Under cygwin the default render for man is
"/usr/bin/less \-isrR" \(see \-P option for man man\)

See sub pagers\_guessing in module Pod&#8203;::Perldoc\.

Workaround&#8203;:
 under the shell\, define PERLDOC\_PAGER="less \-isrR"

What to do do solve this bug?
 Add this block to Perldoc\.pm\, sub pagers\_guessing
&nbsp;    elsif \(IS\_Cygwin\) \{
&nbsp;         push @&#8203;pagers\, '/usr/bin/less \-isrR';
&nbsp;    \}

Olivier Mengué\.



---


---
Flags&#8203;:
&nbsp;    category=utilities
&nbsp;    severity=low


---
Site configuration information for perl v5\.8\.5&#8203;:

Configured by gerrit at Thu Aug 19 19&#8203;:39&#8203;:49     2004\.

Summary of my perl5 \(revision 5 version 8 subversion 5\) configuration&#8203;:
&nbsp;  Platform&#8203;:
&nbsp;    osname=cygwin\, osvers=1\.5\.10\(0\.11642\)\, archname=cygwin\-thread\-multi\-64int
&nbsp;    uname='cygwin\_nt\-5\.1 inspiron 1\.5\.10\(0\.11642\) 2004\-05\-25 22&#8203;:07 i686 unknown
unknown cygwin '
&nbsp;    config\_args='\-de \-Dmksymlinks \-Duse64bitint \-Dusethreads \-Doptimize=\-O2
\-Dman3ext=3pm'
&nbsp;    hint=recommended\, useposix=true\, d\_sigaction=define
&nbsp;    usethreads=define use5005threads=undef useithreads=define
usemultiplicity=define
&nbsp;    useperlio=define d\_sfio=undef uselargefiles=define usesocks=undef
&nbsp;    use64bitint=define use64bitall=undef uselongdouble=undef
&nbsp;    usemymalloc=y\, bincompat5005=undef
&nbsp;  Compiler&#8203;:
&nbsp;    cc='gcc'\, ccflags ='\-DPERL\_USE\_SAFE\_PUTENV \-fno\-strict\-aliasing \-pipe
\-I/usr/local/include'\,
&nbsp;    optimize='\-O2'\,
&nbsp;    cppflags='\-DPERL\_USE\_SAFE\_PUTENV \-fno\-strict\-aliasing \-pipe
\-I/usr/local/include'
&nbsp;    ccversion=''\, gccversion='3\.3\.3 \(cygwin special\)'\, gccosandvers=''
&nbsp;    intsize=4\, longsize=4\, ptrsize=4\, doublesize=8\, byteorder=12345678
&nbsp;    d\_longlong=define\, longlongsize=8\, d\_longdbl=define\, longdblsize=12
&nbsp;    ivtype='long long'\, ivsize=8\, nvtype='double'\, nvsize=8\, Off\_t='off\_t'\,
lseeksize=8
&nbsp;    alignbytes=8\, prototype=define
&nbsp;  Linker and Libraries&#8203;:
&nbsp;    ld='ld2'\, ldflags =' \-s \-L/usr/local/lib'
&nbsp;    libpth=/usr/local/lib /usr/lib /lib
&nbsp;    libs=\-lgdbm \-ldb \-lcrypt \-lgdbm\_compat
&nbsp;    perllibs=\-lcrypt \-lgdbm\_compat
&nbsp;    libc=/usr/lib/libc\.a\, so=dll\, useshrplib=true\, libperl=libperl\.a
&nbsp;    gnulibc\_version=''
&nbsp;  Dynamic Linking&#8203;:
&nbsp;    dlsrc=dl\_dlopen\.xs\, dlext=dll\, d\_dlsymun=undef\, ccdlflags=' \-s'
&nbsp;    cccdlflags=' '\, lddlflags=' \-s \-L/usr/local/lib'

Locally applied patches&#8203;:




---
@&#8203;INC for perl v5\.8\.5&#8203;:
&nbsp;    /usr/lib/perl5/5\.8\.5/cygwin\-thread\-multi\-64int
&nbsp;    /usr/lib/perl5/5\.8\.5
&nbsp;    /usr/lib/perl5/site\_perl/5\.8\.5/cygwin\-thread\-multi\-64int
&nbsp;    /usr/lib/perl5/site\_perl/5\.8\.5
&nbsp;    /usr/lib/perl5/site\_perl
&nbsp;    /usr/lib/perl5/vendor\_perl/5\.8\.5/cygwin\-thread\-multi\-64int
&nbsp;    /usr/lib/perl5/vendor\_perl/5\.8\.5
&nbsp;    /usr/lib/perl5/vendor\_perl
&nbsp;    \.



---
Environment for perl v5\.8\.5&#8203;:
&nbsp;    CYGWIN\_ROOT=D&#8203;:\\Outils\\cygwin
&nbsp;    HOME=/cygdrive/z/MengueO
&nbsp;    LANG=fr\_FR@&#8203;euro
&nbsp;    LANGUAGE \(unset\)
&nbsp;    LD\_LIBRARY\_PATH \(unset\)
&nbsp;    LOGDIR \(unset\)
&nbsp;   
PATH=/opt/qt/3\.2/bin&#8203;:/opt/kde3/bin&#8203;:/opt/kde3/lib&#8203;:/opt/kde3/lib/kde3&#8203;:/usr/local/bin&#8203;:/usr/bin&#8203;:/bin&#8203;:/usr/X11R6/bin&#8203;:\.&#8203;:/usr/bin&#8203;:/usr/X11R6/bin&#8203;:/cygdrive/c/Program
Files/Microsoft
Office/Office&#8203;:/cygdrive/c/WINNT/SYSTEM32&#8203;:/cygdrive/c/WINNT&#8203;:/cygdrive/c/PVCS/DIMENS~1/7\.1/PROG&#8203;:/cygdrive/d/outils/imagemagick&#8203;:/cygdrive/c/WINNT/system32&#8203;:/cygdrive/c/WINNT&#8203;:%InocuLAN%&#8203;:/cygdrive/c/NETMANAG\.32&#8203;:/cygdrive/d/oracle/ora92/bin&#8203;:/cygdrive/c/Program
Files/Oracle/jre/1\.3\.1/bin&#8203;:/cygdrive/c/Program
Files/Oracle/jre/1\.1\.8/bin&#8203;:/cygdrive/d/Outils/Perl/bin&#8203;:/cygdrive/d/Outils/doxygen/bin&#8203;:/cygdrive/c/WINNT/system32/WBEM&#8203;:/cygdrive/z/MengueO/bin&#8203;:/cygdrive/d/Outils/bin&#8203;:/cygdrive/d/Outils/j2sdk1\.4\.2\_03/bin&#8203;:/cygdrive/d/Outils/cscout/bin
&nbsp;    PERL\_BADLANG \(unset\)
&nbsp;    SHELL \(unset\)

**Comment by p5pRT** [comment]
### From @smpeters

> \[dolmen@&#8203;users\.sourceforge\.net \- Fri Oct 15 08&#8203;:32&#8203;:53 2004\]&#8203;:
> 
> This is a bug report for perl from dolmen@&#8203;users\.sourceforge\.net\,
> generated with the help of perlbug 1\.35 running under perl v5\.8\.5\.
> 
> \(but manually sent by email\)
> 
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> 
> 'perldoc \-oman' is incorrectly rendered under Cygwin\.
> I tried with the following terminals&#8203;: cygwin\, xterm\, rxvt\.
> 
> The problem is that the nroff output is piped to the default
> pager which is not defined for cygwin\.
> Under cygwin the default render for man is
> "/usr/bin/less \-isrR" \(see \-P option for man man\)
> 
> See sub pagers\_guessing in module Pod&#8203;::Perldoc\.
> 
> Workaround&#8203;:
>  under the shell\, define PERLDOC\_PAGER="less \-isrR"
> 
> What to do do solve this bug?
>  Add this block to Perldoc\.pm\, sub pagers\_guessing
>     elsif \(IS\_Cygwin\) \{
>          push @&#8203;pagers\, '/usr/bin/less \-isrR';
>     \}
> 
> Olivier Mengué\.
> 

I could not get your change to work because the pager "/usr/bin/less" is
first in the @&#8203;pagers array and used\.  I did\, however\, get it to work
with "unshift @&#8203;pagers\, '/usr/bin/less \-isrR';"  It does make nice\, easy
to read perldoc's on Cygwin\, so it'd be great to get this in\.  A patch
is included\.

Steve Peters

**Comment by p5pRT** [comment]
### From @smpeters

<details><summary>perldoc.patch</summary>

```diff
--- Perldoc.pm.orig	2004-10-19 08:52:58.635955200 -0500
+++ Perldoc.pm	2004-10-19 09:09:24.183102400 -0500
@@ -1225,6 +1225,10 @@
         push @pagers, qw( less.exe more.com< );
         unshift @pagers, $ENV{PAGER}  if $ENV{PAGER};
     }
+    elsif (IS_Cygwin) {
+        unshift @pagers, '/usr/bin/less -isrR';
+        unshift @pagers, $ENV{PAGER} if $ENV{PAGER};
+    }
     else {
         if (IS_OS2) {
           unshift @pagers, 'less', 'cmd /c more <';

```
</details>

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @smpeters

> \[stmpeters \- Tue Oct 19 07&#8203;:13&#8203;:13 2004\]&#8203;:
> 
> > \[dolmen@&#8203;users\.sourceforge\.net \- Fri Oct 15 08&#8203;:32&#8203;:53 2004\]&#8203;:
> > 
> > This is a bug report for perl from dolmen@&#8203;users\.sourceforge\.net\,
> > generated with the help of perlbug 1\.35 running under perl v5\.8\.5\.
> > 
> > \(but manually sent by email\)
> > 
> > \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> > 
> > 'perldoc \-oman' is incorrectly rendered under Cygwin\.
> > I tried with the following terminals&#8203;: cygwin\, xterm\, rxvt\.
> > 
> > The problem is that the nroff output is piped to the default
> > pager which is not defined for cygwin\.
> > Under cygwin the default render for man is
> > "/usr/bin/less \-isrR" \(see \-P option for man man\)
> > 
> > See sub pagers\_guessing in module Pod&#8203;::Perldoc\.
> > 
> > Workaround&#8203;:
> >  under the shell\, define PERLDOC\_PAGER="less \-isrR"
> > 
> > What to do do solve this bug?
> >  Add this block to Perldoc\.pm\, sub pagers\_guessing
> >     elsif \(IS\_Cygwin\) \{
> >          push @&#8203;pagers\, '/usr/bin/less \-isrR';
> >     \}
> > 
> > Olivier Mengué\.
> > 
> 
> I could not get your change to work because the pager "/usr/bin/less" is
> first in the @&#8203;pagers array and used\.  I did\, however\, get it to work
> with "unshift @&#8203;pagers\, '/usr/bin/less \-isrR';"  It does make nice\, easy
> to read perldoc's on Cygwin\, so it'd be great to get this in\.  A patch
> is included\.
> 
> Steve Peters
> 

This problem has been fixed in Pod&#8203;::Perldoc 3\.14\.  Can this version
please be integrated in?

Thanks in advance\!

**Comment by p5pRT** [comment]
### From @smpeters

> \[stmpeters \- Wed Dec 01 12&#8203;:20&#8203;:44 2004\]&#8203;:
> 
> > \[stmpeters \- Tue Oct 19 07&#8203;:13&#8203;:13 2004\]&#8203;:
> > 
> > > \[dolmen@&#8203;users\.sourceforge\.net \- Fri Oct 15 08&#8203;:32&#8203;:53 2004\]&#8203;:
> > > 
> > > This is a bug report for perl from dolmen@&#8203;users\.sourceforge\.net\,
> > > generated with the help of perlbug 1\.35 running under perl v5\.8\.5\.
> > > 
> > > \(but manually sent by email\)
> > > 
> > > \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> > > 
> > > 'perldoc \-oman' is incorrectly rendered under Cygwin\.
> > > I tried with the following terminals&#8203;: cygwin\, xterm\, rxvt\.
> > > 
> > > The problem is that the nroff output is piped to the default
> > > pager which is not defined for cygwin\.
> > > Under cygwin the default render for man is
> > > "/usr/bin/less \-isrR" \(see \-P option for man man\)
> > > 
> > > See sub pagers\_guessing in module Pod&#8203;::Perldoc\.
> > > 
> > > Workaround&#8203;:
> > >  under the shell\, define PERLDOC\_PAGER="less \-isrR"
> > > 
> > > What to do do solve this bug?
> > >  Add this block to Perldoc\.pm\, sub pagers\_guessing
> > >     elsif \(IS\_Cygwin\) \{
> > >          push @&#8203;pagers\, '/usr/bin/less \-isrR';
> > >     \}
> > > 
> > > Olivier Mengué\.
> > > 
> > 
> > I could not get your change to work because the pager "/usr/bin/less" is
> > first in the @&#8203;pagers array and used\.  I did\, however\, get it to work
> > with "unshift @&#8203;pagers\, '/usr/bin/less \-isrR';"  It does make nice\, easy
> > to read perldoc's on Cygwin\, so it'd be great to get this in\.  A patch
> > is included\.
> > 
> > Steve Peters
> > 
> 
> This problem has been fixed in Pod&#8203;::Perldoc 3\.14\.  Can this version
> please be integrated in?
> 
> Thanks in advance\!
> 

Pod&#8203;::Perldoc 3\.14 integrated with change 23630\.  Thanks\!

**Comment by p5pRT** [comment]
@smpeters - Status changed from 'open' to 'resolved'
